contrib/spring_cloud/core/src/main/java/io/opencensus/contrib/springcloud/instrument/web/HttpServletFilter.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.springcloud.instrument.web;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.opencensus.contrib.http.servlet.OcHttpServletFilter;
+import io.opencensus.contrib.springcloud.autoconfig.OpenCensusAutoConfiguration;
+import javax.servlet.Filter;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(OpenCensusAutoConfiguration.TRACE_FILTER_ORDER)
+@SuppressFBWarnings("RI_REDUNDANT_INTERFACES")
+public class HttpServletFilter extends OcHttpServletFilter implements Filter {}
```

Comment:
An alternative to using the Servet filter would be to use a Spring MVC interceptor and the `WebMvcConfigurer`. We did this for including the trace id in the logs. The same concept applies to this PR. Take a look in [our code here](https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/logging/TraceIdLoggingWebMvcInterceptor.java).

Reply:
thanks for the review. Is Servlet filter approach still ok? The reason I asked is the most of the work for servlet filter is done and usable outside of the spring frame work as well. But if there are advantages of doing it using MVC interceptor i am can certainly change it.

Reply:
Sure, Servlet Filter is more generic.